File: leetcode/243e.py
"""
Given an array of strings wordsDict and two different strings that already exist in the array word1 and word2, return the shortest distance between these two words in the list.
"""
class Solution:
    def shortestDistance(self, wordsDict: List[str], word1: str, word2: str) -> int:
        poss1 = [idx for idx, word in enumerate(wordsDict) if word == word1]
        poss2 = [idx for idx, word in enumerate(wordsDict) if word == word2]

        min_distance = float('inf')
        for pos1 in poss1:
            for pos2 in poss2:
                min_distance = min(min_distance, abs(pos1 - pos2))
        return min_distance

File: leetcode/243e.java
class Solution {
    public int shortestDistance(String[] wordsDict, String word1, String word2) {
        List<Integer> word1Indices = new ArrayList<Integer>();
        List<Integer> word2Indices = new ArrayList<Integer>();

        for (int i = 0; i < wordsDict.length; i++) {
            String word = wordsDict[i];
            if (word.equals(word1)) {
                word1Indices.add(i);
            } else if (word.equals(word2)) {
                word2Indices.add(i);
            }
        }

        int shortestDistance = 99999999;
        for (int i = 0; i < word1Indices.size(); i++) {
            for (int j = 0; j < word2Indices.size(); j++) {
                shortestDistance = Math.min(shortestDistance, Math.abs(word1Indices.get(i) - word2Indices.get(j)));
            }
        }

        return shortestDistance;
    }
}
